Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -362,60 +414,73 @@ void FieldProbe::ComputeDiags (int step)
-                     *  [Ex, Ey, Ez, Bx, By, Bz, and S] */
+                /* m_data_vector now contains up-to-date values for:
+                 *  [i, Rx, Ry, Rz, Ex, Ey, Ez, Bx, By, Bz, and S] */
+                m_data_vector.resize(np * noutputs);
                 }
-
-                // do we have the one and only probe particle on our MPI rank?
-                if (np > 0)
-                    probe_proc = amrex::ParallelDescriptor::MyProc();
             }
 
         } // end particle iterator loop
+//mpi gather.... amrex::ParallelGather
 
         // make sure data is in m_data
         Gpu::synchronize();
-
-        // this check is here because for m_field_probe_integrate == True, we always compute
-        // but we only write when we truly are in an output interval step
-        if (m_intervals.contains(step+1)) {
-            /*
-             * All the processors have probe_proc = -1 except the one that contains the point, which
-             * has probe_proc equal to a number >=0. Therefore, ReduceIntMax communicates to all the
-             * processors the rank of the processor which contains the point
-             */
-            amrex::ParallelDescriptor::ReduceIntMax(probe_proc);
-
-            if (probe_proc != amrex::ParallelDescriptor::IOProcessorNumber() and probe_proc != -1) {
-                if (amrex::ParallelDescriptor::MyProc() == probe_proc) {
-                    amrex::ParallelDescriptor::Send(m_data.data(), noutputs,
-                                                    amrex::ParallelDescriptor::IOProcessorNumber(),
-                                                    0);
-                }
-                if (amrex::ParallelDescriptor::MyProc()
-                    == amrex::ParallelDescriptor::IOProcessorNumber()) {
-                    amrex::ParallelDescriptor::Recv(m_data.data(), noutputs, probe_proc, 0);
-                }
-            }
-        } // send to IO Processor
+        m_data_out.resize(m_probe.TotalNumberOfParticles() * noutputs);
+        amrex::ParallelGather::Gather (m_data_vector.data(), m_data_vector.size(), m_data_out.data(), amrex::ParallelDescriptor::IOProcessorNumber(), amrex::ParallelDescriptor::Communicator());
     }// end loop over refinement levels
 } // end void FieldProbe::ComputeDiags
 
 void FieldProbe::WriteToFile (int step) const
 {
     if (ProbeInDomain() && amrex::ParallelDescriptor::IOProcessor())
     {
-        ReducedDiags::WriteToFile (step);
+        // open file
+        std::ofstream ofs{m_path + m_rd_name + "." + m_extension,
+               std::ofstream::out | std::ofstream::app};
+
+        // write step
+        ofs << step+1;
+
+        ofs << m_sep;
+
+        // set precision
+        ofs << std::fixed << std::setprecision(14) << std::scientific;
+
+        // write time
+        ofs << WarpX::GetInstance().gett_new(0);
+
+        // loop over data size and write
+        for (int i = 0; i < m_probe.TotalNumberOfParticles(); i++)
```

Comment:
This probably won't work because it's inside amrex::ParallelDescriptor::IOProcessor().

Reply:
Correct, please use `m_probe.size() / noutputs` instead.